app/models/discussion.rb

```diff
@@ -65,6 +65,17 @@ class Discussion < ActiveRecord::Base
   pg_search_scope :search, against: [:title, :description],
     using: {tsearch: {dictionary: "english"}}
 
+
+  scope :search_for, ->(query, user, opts = {}) do
```

Comment:
So, searching for discussions has been reduced to 3 scopes, one on Discussion and two on SearchVector

- SearchVector#search_without_privacy returns all of the matches across the system, and returns their discussion_id and search rank
- SearchVector#search_for applies the Queries::VisibleDiscussions privacy option
- Discussion#search_for takes all of the discussion ids in SearchVector#search_for, and selects the fields we're interested in (title, description, key, a blurb, etc.) based on them.

Note that these queries are all knit together in ActiveRecord, so it all happens in one query that's pretty darn fast.